Index: bas-cfengine-debian/cf-agent/verify_files_utils.c
===================================================================
--- bas-cfengine-debian.orig/cf-agent/verify_files_utils.c	2013-10-28 19:59:50.274513292 +0100
+++ bas-cfengine-debian/cf-agent/verify_files_utils.c	2013-10-28 20:00:36.768558703 +0100
@@ -671,6 +671,8 @@
             char backup[CF_BUFSIZE];
             mode_t mask;
 
+            if (attr.copy.type_check)
+            {
             if (!attr.move_obstructions)
             {
                 Log(LOG_LEVEL_INFO, "Path '%s' is a symlink. Unable to move it aside without move_obstructions is set",
@@ -696,6 +698,24 @@
                 return;
             }
             umask(mask);
+            }
+            else
+            {
+                struct stat tmpstat; 
+
+                if (stat(to, &tmpstat) != 0)
+                {
+                    cfPS(ctx, LOG_LEVEL_ERR, PROMISE_RESULT_WARN, pp, attr, "Unable to stat newly created directory '%s'. (stat: %s)",
+                         to, GetErrorStr());
+                    return PROMISE_RESULT_WARN;
+                }
+
+                if (!S_ISDIR(tmpstat.st_mode))
+                {
+                    cfPS(ctx, LOG_LEVEL_ERR, PROMISE_RESULT_WARN, pp, attr, "Symlink does not point to a directory: '%s'", to);
+                    return PROMISE_RESULT_WARN;
+                }
+            }
         }
     }
 
