Index: Cfengine-debian/configure.ac
===================================================================
--- Cfengine-debian.orig/configure.ac	2013-10-30 12:03:16.825233700 +0100
+++ Cfengine-debian/configure.ac	2013-10-30 12:05:40.146134357 +0100
@@ -122,7 +122,6 @@
 
 AS_IF([test x"$enable_fhs" = xyes], [
   projlibdir='${libdir}/cfengine'
-  projdatadir='${exec_prefix}/share/cfengine'
   WORKDIR='${localstatedir}/lib/cfengine'
   LOGDIR='${localstatedir}/lib/cfengine'
   PIDDIR=='${localstatedir}/lib/cfengine'
@@ -149,12 +148,10 @@
 
   sbindir='${exec_prefix}/bin' # /var/cfengine/bin despite being sbin_?
   projlibdir='${exec_prefix}/lib'
-  projdatadir='${exec_prefix}/share'
   mandir='${exec_prefix}/share/man'
 ])
 
 AC_SUBST(projlibdir)
-AC_SUBST(projdatadir)
 
 dnl ######################################################################
 dnl Enable debugging
Index: Cfengine-debian/masterfiles/Makefile.am
===================================================================
--- Cfengine-debian.orig/masterfiles/Makefile.am	2013-10-30 12:07:39.606219125 +0100
+++ Cfengine-debian/masterfiles/Makefile.am	2013-10-30 12:08:07.223163441 +0100
@@ -1,4 +1,4 @@
-masterfilesdir = $(projdatadir)/CoreBase
+masterfilesdir = $(datadir)/CoreBase
 dist_masterfiles_DATA = $(srcdir)/promises.cf $(srcdir)/def.cf $(srcdir)/cf-sketch-runfile.cf $(srcdir)/update.cf
 
 #failsafedir = $(masterfilesdir)/failsafe
