diff --git a/Makefile.am b/Makefile.am
index 0b6d5e8..611791d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -25,7 +25,7 @@ DIST_COMMON = README.md Makefile.am Makefile.in configure AUTHORS aclocal.m4 \
 
 EXTRA_DIST = ChangeLog INSTALL README.md LICENSE
 
-docsdir = $(projdocdir)
+docsdir = $(docdir)
 docs_DATA = README.md ChangeLog
 
 #
diff --git a/configure.ac b/configure.ac
index c00a6a3..1eb7a0b 100755
--- a/configure.ac
+++ b/configure.ac
@@ -137,7 +137,6 @@ AC_ARG_ENABLE([fhs],
 AS_IF([test x"$enable_fhs" = xyes], [
   projlibdir='${libdir}/cfengine'
   projdatadir='${exec_prefix}/share/cfengine'
-  projdocdir='${exec_prefix}/share/doc/cfengine'
   WORKDIR='${localstatedir}/lib/cfengine'
   LOGDIR='${localstatedir}/lib/cfengine'
   PIDDIR=='${localstatedir}/lib/cfengine'
@@ -165,13 +164,11 @@ AS_IF([test x"$enable_fhs" = xyes], [
   sbindir='${exec_prefix}/bin' # /var/cfengine/bin despite being sbin_?
   projlibdir='${exec_prefix}/lib'
   projdatadir='${exec_prefix}/share'
-  projdocdir='${exec_prefix}/share/doc'
   mandir='${exec_prefix}/share/man'
 ])
 
 AC_SUBST(projlibdir)
 AC_SUBST(projdatadir)
-AC_SUBST(projdocdir)
 
 dnl ######################################################################
 dnl Enable debugging
diff --git a/examples/Makefile.am b/examples/Makefile.am
index 31d7bcf..8cc08ee 100644
--- a/examples/Makefile.am
+++ b/examples/Makefile.am
@@ -1,4 +1,4 @@
-examplesdir = $(projdocdir)/examples
+examplesdir = $(docdir)/examples
 dist_examples_DATA = $(srcdir)/*.cf
 
 MAINTAINERCLEANFILES = Makefile.in mdate-sh
