--- a/Makefile.am
+++ b/Makefile.am
@@ -26,7 +26,7 @@
 
 EXTRA_DIST = ChangeLog INSTALL README.md LICENSE
 
-docsdir = $(projdocdir)
+docsdir = $(docdir)
 docs_DATA = README.md ChangeLog
 
 #
--- a/configure.ac
+++ b/configure.ac
@@ -123,7 +123,6 @@
 AS_IF([test x"$enable_fhs" = xyes], [
   projlibdir='${libdir}/cfengine'
   projdatadir='${exec_prefix}/share/cfengine'
-  projdocdir='${exec_prefix}/share/doc/cfengine'
   WORKDIR='${localstatedir}/lib/cfengine'
   LOGDIR='${localstatedir}/lib/cfengine'
   PIDDIR=='${localstatedir}/lib/cfengine'
@@ -151,13 +150,11 @@
   sbindir='${exec_prefix}/bin' # /var/cfengine/bin despite being sbin_?
   projlibdir='${exec_prefix}/lib'
   projdatadir='${exec_prefix}/share'
-  projdocdir='${exec_prefix}/share/doc'
   mandir='${exec_prefix}/share/man'
 ])
 
 AC_SUBST(projlibdir)
 AC_SUBST(projdatadir)
-AC_SUBST(projdocdir)
 
 dnl ######################################################################
 dnl Enable debugging
--- a/examples/Makefile.am
+++ b/examples/Makefile.am
@@ -1,4 +1,4 @@
-examplesdir = $(projdocdir)/examples
+examplesdir = $(docdir)/examples
 dist_examples_DATA = $(srcdir)/*.cf
 
 MAINTAINERCLEANFILES = Makefile.in mdate-sh
