Index: orig/tests/acceptance/01_vars/02_functions/strftime.cf
===================================================================
--- orig.orig/tests/acceptance/01_vars/02_functions/strftime.cf	2013-10-13 01:14:47.087793198 +0200
+++ orig/tests/acceptance/01_vars/02_functions/strftime.cf	2013-10-13 11:57:29.468260936 +0200
@@ -35,7 +35,7 @@
     Saturday::
       "dow" string => "6";
     Sunday::
-      "dow" string => "7";
+      "dow" string => "0";
 
   files:
       "$(G.testfile).expected"
